src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ValidationHelpers.cs

```diff
@@ -90,7 +89,7 @@ public static string GetModelErrorMessageOrDefault(ModelError modelError)
             ModelMetadata metadata,
             List<ModelStateEntry> orderedModelStateEntries)
         {
-            if (metadata.ElementMetadata != null)
+            if (metadata.ElementMetadata != null && modelStateEntry.Children != null)
```

Comment:
:see_no_evil: 

We have no numbers justifying perf-related changes in this method and this is not a likely hot path.